### PR TITLE
feat(connector): [Coinbase] Add order id as the reference id 

### LINKED ISSUE #2326: [FEATURE]: [Coinbase] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Ceres445**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Relevant documentation [here](https://docs.cloud.coinbase.com/commerce/docs/webhooks-events). I'm not 100% sure what `code` is in the documentation but they haven't provided more clarity, the only other identifier for the transaction is id, so I've used that.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Fixes #2326 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/coinbase/transformers.rs [hunk 1]
```diff
@@ -155,7 +155,7 @@ impl<F, T>
                 message: "Please check the transaction in coinbase dashboard and resolve manually"
                     .to_string(),
                 }),
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(item.response.data.id.clone()),
```
Comment (srujanchikke) [inline crates/router/src/connector/coinbase/transformers.rs:158]: Hi @Ceres445 ,

`clone` is not required here .

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by ArjunKarthik — APPROVED** [review]

**Comment by ArjunKarthik** [comment]
@Ceres445 Thanks for the PR

**Comment by swangi-kumari** [comment]
Hey @Ceres445 
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**